src/compiler/program.ts

```diff
@@ -1952,25 +1949,38 @@ namespace ts {
                 }
             }
 
-            function collectDynamicImportOrRequireCalls(node: Node): void {
-                if (isRequireCall(node, /*checkArgumentIsStringLiteralLike*/ true)) {
-                    imports = append(imports, node.arguments[0]);
-                }
-                // we have to check the argument list has length of 1. We will still have to process these even though we have parsing error.
-                else if (isImportCall(node) && node.arguments.length === 1 && isStringLiteralLike(node.arguments[0])) {
-                    imports = append(imports, node.arguments[0] as StringLiteralLike);
-                }
-                else if (isLiteralImportTypeNode(node)) {
-                    imports = append(imports, node.argument.literal);
-                }
-                collectDynamicImportOrRequireCallsForEachChild(node);
-                if (hasJSDocNodes(node)) {
-                    forEach(node.jsDoc, collectDynamicImportOrRequireCallsForEachChild);
+            function collectDynamicImportOrRequireCalls(file: SourceFile) {
+                const r = /import|require/g;
+                while (r.exec(file.text) !== null) {
+                    const node = getTokenAtPosition(file, r.lastIndex);

```

Comment:
as this also includes trivia, you will add the following imports twice to the list:

```ts
/* using require() */ require('foo');
/* using import() */ import('bar');
```

Reply:
For the comment-internal `require`, we end up with the Identifier require, not the CallExpression require. So we don't end up double-adding imports.

Reply:
I realized that too after writing the comment.
Though there's still the double-adding when the comment is at another position:
```ts
require/* using require() */ ('foo');
```

Reply:
Good catch. I added a check that the regex's end must be the same as the retrieved node's end.

Reply:
Actually, @weswigham points out that duplicates are fine -- it's just like writing `import('bar')` twice.